tests/cephfs/snapshot_clone/cephfs_snap_utils.py

```diff
@@ -0,0 +1,435 @@
+"""
+This is cephfs snapshot schedule utility module
+It contains all the re-useable functions related to cephfs snapshot schedule and retention feature
+
+"""
+import datetime
+import json
+import random
+import re
+import string
+import time
+
+from ceph.ceph import CommandFailed
+from utility.log import Log
+from utility.retry import retry
+
+log = Log(__name__)
+
+
+class SnapUtils(object):
+    def __init__(self, ceph_cluster):
+        """
+        FS Snapshot Utility object
+        Args:
+            ceph_cluster (ceph.ceph.Ceph): ceph cluster
+        """
+
+        self.ceph_cluster = ceph_cluster
+        self.mons = ceph_cluster.get_ceph_objects("mon")
+        self.mgrs = ceph_cluster.get_ceph_objects("mgr")
+        self.osds = ceph_cluster.get_ceph_objects("osd")
+        self.mdss = ceph_cluster.get_ceph_objects("mds")
+
+    def enable_schedule(self, client):
+        """
+        Enables Snapshot schedule on Ceph
+        Args:
+            client: ceph client to run cmd
+        Returns: None
+        """
+        out, rc = client.exec_command(
+            sudo=True, cmd="ceph mgr module enable snap_schedule"
+        )
+        log.info(out)
+
+    def modify_min_schedule(self, client, allow=True):
+        """
+        Modifies ceph config to allow minutely snapshots in snapshot schedule
+        Args:
+            client: ceph client to run cmd
+            allow: Boolean option to allow modify or not, True|False , default=True
+        Returns: None
+        """
+        out, rc = client.exec_command(
+            sudo=True,
+            cmd=f"ceph config set mgr mgr/snap_schedule/allow_m_granularity {allow}",
+        )
+        log.info(out)
+
+    def create_snap_schedule(self, snap_params):
+        """
+        To create snapshot schedule on ceph FS and activate it. It also verifies that schedule is set.
+        Args:
+            client: ceph client to run cmd
+            snap_params : a dict data type with below key-value pairs,
+                Required:
+                    path : a path for scheduled snapshot to be created, type - str
+                    sched : a schedule to be applied to snapshot schedule ass, type - str, for eg., 2M1h
+                    fs_name : cephfs volume name
+                Optional:
+                    start_time : a iso 8601 format time value, from when schedule is to be applied. type -str
+        Returns: 0 : success, 1 : failure
+        """
+        client = snap_params["client"]
+        sched_cmd = (
+            f"ceph fs snap-schedule add {snap_params['path']} {snap_params['sched']}"
+        )
+        if snap_params.get("start_time"):
+            sched_cmd += f" {snap_params['start_time']} "
+        sched_cmd += f" --fs {snap_params['fs_name']}"
+
+        activate_cmd = f"ceph fs snap-schedule activate {snap_params['path']} --fs {snap_params['fs_name']}"
```

Comment:
I don't think activating snapshot_schedule is required after adding.
When you add snap-schedule - it automatically enables the schedules.

Reply:
let's have a different function for activating schedules also. Can be used to validate enabling and disabling schedules later